contracts/deploy/plume/005_rooster_amo.js

```diff
@@ -0,0 +1,148 @@
+const {
+  deployWithConfirmation,
+  withConfirmation,
+} = require("../../utils/deploy");
+const addresses = require("../../utils/addresses");
+const {
+  deployPlumeRoosterAMOStrategyImplementation,
+} = require("../deployActions");
+const { isFork, oethUnits } = require("../../test/helpers");
+const { setERC20TokenBalance } = require("../../test/_fund");
+const { utils } = require("ethers");
+
+module.exports = deployOnPlume(
+  {
+    deployName: "005_rooster_amo",
+  },
+  async () => {
+    const { deployerAddr } = await getNamedAccounts();
+    const sDeployer = await ethers.getSigner(deployerAddr);
+    const cOETHpVaultProxy = await ethers.getContract("OETHPlumeVaultProxy");
+    const weth = await ethers.getContractAt("IWETH9", addresses.plume.WETH);
+    const deployerWethBalance = await weth
+      .connect(sDeployer)
+      .balanceOf(sDeployer.address);
+
+    console.log("Deployer WETH balance", deployerWethBalance.toString());
+    if (!isFork) {
+      if (deployerWethBalance.lt(oethUnits("1"))) {
+        throw new Error(
+          "Deployer needs at least 1e18 of WETH to mint the initial balance"
+        );
+      }
+    }
+
+    const cOETHpVault = await ethers.getContractAt(
+      "IVault",
+      cOETHpVaultProxy.address
+    );
+
+    await deployWithConfirmation("RoosterAMOStrategyProxy");
+    const cAMOStrategyProxy = await ethers.getContract(
+      "RoosterAMOStrategyProxy"
+    );
+
+    if (isFork) {
+      // Just pretend wPlume is the reward token for testing
+      await deployWithConfirmation("MockMaverickDistributor", [
+        addresses.plume.WPLUME,
+      ]);
+      const cMockMaverickDistributor = await ethers.getContract(
+        "MockMaverickDistributor"
+      );
+
+      await withConfirmation(
+        cMockMaverickDistributor
+          .connect(sDeployer)
+          .setRewardTokenAmount(oethUnits("1"))
+      );
+
+      // Fund the mock contract
+      const wPlume = await ethers.getContractAt(
+        "IWETH9",
+        addresses.plume.WPLUME
+      );
+      await withConfirmation(
+        wPlume.connect(sDeployer).deposit({ value: oethUnits("10") })
+      );
+      await withConfirmation(
+        wPlume
+          .connect(sDeployer)
+          .transfer(cMockMaverickDistributor.address, oethUnits("10"))
+      );
+    }
+
+    const cAMOStrategyImpl = await deployPlumeRoosterAMOStrategyImplementation(
+      addresses.plume.OethpWETHRoosterPool
+    );
+    const strategyImplInitData = cAMOStrategyImpl.interface.encodeFunctionData(
+      "initialize()",
+      []
+    );
+
+    // prettier-ignore
+    await withConfirmation(
+      cAMOStrategyProxy
+        .connect(sDeployer)["initialize(address,address,bytes)"](
+          cAMOStrategyImpl.address,
+          addresses.plume.timelock,
+          strategyImplInitData
+        )
+    );
+
+    const cAMOStrategy = await ethers.getContractAt(
+      "RoosterAMOStrategy",
+      cAMOStrategyProxy.address
+    );
+
+    if (isFork) {
```

Comment:
Is there a reason for doing things differently in the fork vs the real deploy? This means the real deploy settings won't have been actually tested in the fork.

Reply:
agreed, thanks: https://github.com/OriginProtocol/origin-dollar/pull/2499/commits/3640538075eb907172b07b124089a0c8cb3292a1